shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/STS/Utxow.hs

```diff
@@ -160,10 +171,11 @@ utxoWitnessed =
         ?! MissingScriptWitnessesUTXOW
 
       -- check VKey witnesses
-      verifiedWits tx ?! InvalidWitnessesUTXOW
+      let failedWits = verifiedWits tx
+      failedWits == mempty ?! InvalidWitnessesUTXOW failedWits
```

Comment:
Note that `Either [..] ()` will let you use `?!:` :-D

Reply:
it's like Christmas in May! :christmas_tree:

Reply:
(seriously, that's great)